### PR TITLE
Docs: Adds troubleshooting for test flag

### LINKED ISSUE #31: [Bug] Doesn't work when Storybook is built with the new --test flag
### Describe the bug

Storybook 7.6 introduced a new --test flag for building Storybook with less of the fluff that is not needed for testing environments.
When you try to run test-storybook with --coverage on a storybook that was built with --test, I get the following error:
```
The code in this story is not instrumented, which means the coverage setup is likely not correct.
```

### Steps to reproduce the behavior

1. Build a Storybook with --test
2. Run test-storybook on that with --coverage
3. Rebuild the same Storybook without --test
4. Test storybook passes

### Expected behavior

I believe that the istanbul plugin is not being properly injected when --test is enabled. I don't get the `[addon-coverage] Adding istanbul plugin to Vite config` message when I use it. The coverage should work when --test is enabled.

### PR DESCRIPTION
**Pull request by jonniebigodes**
With this pull request, the README file was updated to factor in a troubleshooting section to address and close #31 until a proper fix is provided.

What was done:
- Polished the documentation
- Added a Troubleshooting section to factor in the issue

@yannbf  or @kasperpeulen, when you have a moment, can you take a look at this and follow up with me on it? Thanks in advance

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -77,12 +77,42 @@ export default {
...
+    '@storybook/addon-links',
+    '@storybook/addon-essentials',
+    '@storybook/addon-interactions',
+    '@storybook/addon-coverage',
+  ],
+  build: {
+    test: {
+      disabledAddons: [
+        '@storybook/addon-docs',
+        '@storybook/addon-essentials/docs',
```
Comment (yannbf) [inline README.md:109]: is this needed in the snippet?

File: README.md [hunk 2]
```diff
@@ -77,12 +77,42 @@ export default {
...
+
+```js
+// main.js
+
+export default {
+  addons: [
+    '@storybook/addon-links',
+    '@storybook/addon-essentials',
+    '@storybook/addon-interactions',
+    '@storybook/addon-coverage',
```
Comment (yannbf) [inline README.md:103]: do we need to add the whole list, or the addons field at all?

**Review by yannbf — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (jonniebigodes) [reply README.md:103]: I can tweak the addons—no problem at all there. I just copied them over from an existing reproduction.

File: README.md (hunk 1 above)
Comment (jonniebigodes) [reply README.md:109]: Yes, based on the comment from Shilman. I tested it out and was able to see the issue and the fix.

**Pull request merged**